flatpak_builder_lint/ostree.py

```diff
@@ -1,40 +1,32 @@
-import errno
 import json
 import os
-import subprocess
-from typing import Optional, TypedDict
+from typing import Optional
 
-from .builddir import parse_metadata
+import gi
 
+gi.require_version("OSTree", "1.0")
+from gi.repository import Gio, GLib, OSTree  # noqa: E402
 
-class CliResult(TypedDict):
-    stdout: str
-    stderr: str
-    returncode: int
 
+def open_ostree_repo(repo_path: str) -> OSTree.Repo:
+    repo = OSTree.Repo.new(Gio.File.new_for_path(repo_path))
 
-def cli(repo: str, *args: str) -> CliResult:
-    if not os.path.exists(repo):
-        raise OSError(errno.ENOENT, f"No such ostree repo: {repo}")
+    if not repo.open(None):
+        raise Exception("Failed to open the OSTree repository")
 
-    ret = subprocess.run(
-        ["ostree", f"--repo={repo}", *args],
-        capture_output=True,
-    )
+    return repo
 
-    return {
-        "stdout": ret.stdout.decode("utf-8"),
-        "stderr": ret.stderr.decode("utf-8"),
-        "returncode": ret.returncode,
-    }
 
+def get_refs(repo_path: str, ref_prefix: str | None) -> set:
```

Comment:
Minor nit, you have functions returning a bare `set` and `dict`, instead of specifying the full type.